norddrop/src/ffi/mod.rs

```diff
@@ -675,6 +675,53 @@ pub unsafe extern "C" fn norddrop_new(
     result.unwrap_or(norddrop_result::NORDDROP_RES_ERROR)
 }
 
+/// Set state for multiple peers at once
+///
+/// # Arguments
+///
+/// * `dev` - A pointer to the instance.
+/// * `states` - JSON of states.
+///
+/// # States format
+///
+/// ```json
+/// [
+///   {
+///     "1.2.3.4": "online",
+///   },
+///   {
+///     "5.6.7.8": "offline",
+///   }
+/// ]
+/// ```
+/// # Safety
+/// The pointers provided should be valid
+#[no_mangle]
+pub unsafe extern "C" fn norddrop_set_peers_state(
+    dev: &norddrop,
+    states: *const c_char,
+) -> norddrop_result {
+    let result = panic::catch_unwind(move || {
+        let mut dev = match dev.0.lock() {
+            Ok(inst) => inst,
+            Err(poisoned) => poisoned.into_inner(),
+        };
+
+        let states = {
+            if states.is_null() {
+                return norddrop_result::NORDDROP_RES_INVALID_STRING;
+            }
+
+            ffi_try!(CStr::from_ptr(states).to_str())
```

Comment:
```suggestion
            Str::from_ptr(states).to_str()?
```
There is an implementation `impl From<std::str::Utf8Error> for norddrop_result`